pkg/core/chain/chain.go

```diff
@@ -64,8 +64,8 @@ type Loader interface {
 
 // Ledger is the Chain interface used in tests.
 type Ledger interface {
-	TryNextConsecutiveBlockInSync(block.Block) error
-	TryNextConsecutiveBlockOutSync(block.Block) error
+	TryNextConsecutiveBlockInSync(blk block.Block, kadcastHeight byte) error
+	TryNextConsecutiveBlockOutSync(blk block.Block, kadcastHeight byte) error
```

Comment:
What exactly is this kadcast height for? I am a little bit fuzzy on the paper at this point and I can't really figure out the use of this from the code.

Reply:
`kadcastHeight`  (a.k.a `BCAST_HEIGHT` or `height` ) is kind of an identifier of a subtree in the binary routing tree (Kademlia). It should be propagated with each Kadcast wire message so that the receiver node knows which subtree  the message should re-propagated to.  The broadcast algorithm is well-illustrated at `figure-2` (https://eprint.iacr.org/2019/876.pdf).


NB. I don't like the need of pushing a transport-related field to the upper layers either. But this approach is stateless, lite and compliant with Kadcast protocol. Later on, we might consider a stateful solution where `kadcast Height` is known only by P2P layer.

Reply:
I see. The reason I asked as well, is because I don't really see it mutated anywhere, and it seems that, whenever you need it, you just use the pre-configured initial kadcast height. Is this due to be changed? If so, may be worthwhile to log that in an issue.

Reply:
The kadcast height is decremented in `(w *Writer) broadcastPacket` on re-propagating a message. The Initial Kadcast height is only used when a node triggers a broadcast algorithm initially.

Reply:
Gotcha, makes sense.